mirascope/cohere/types.py

```diff
@@ -288,3 +291,41 @@ def tool_calls(self) -> Optional[list[ToolCall]]:
         if isinstance(self.chunk, StreamedChatResponse_ToolCallsGeneration):
             return self.chunk.tool_calls
         return None
+
+
+class CohereEmbeddingResponse(BaseEmbeddingResponse[SkipValidation[EmbedResponse]]):
+    """A convenience wrapper around the Cohere `EmbedResponse` response."""
+
+    embedding_type: Optional[
+        Literal["float", "int8", "uint8", "binary", "ubinary"]
+    ] = None
+
+    @property
+    def embeddings(
+        self,
+    ) -> Optional[Union[list[list[float]], list[list[int]]]]:
+        """Returns the embeddings"""
+        if self.response.response_type == "embeddings_floats":
+            return self.response.embeddings
+        else:
+            embedding_type = self.embedding_type
+            if embedding_type == "float":
+                embedding_type == "float_"
+
+            # TODO: Update to model_dump when Cohere updates to Pydantic v2
+            embeddings_by_type: EmbedByTypeResponseEmbeddings = self.response.embeddings
+            embedding_dict = embeddings_by_type.dict()
+            return embedding_dict.get(str(embedding_type), None)
```

Comment:
why not just `return self.response.embeddings.dict().get(embedding_type, None)`?

Do we have to convert to string if it's literal and not None? Also how do we handle None?

Reply:
mostly mypy bc embedding_type is a Literal and get expects a str